File: ruby/easy/license_key_formatting.rb
# @param {String} s
# @param {Integer} k
# @return {String}
def license_key_formatting(s, k)
    key, res = s.gsub('-', ''), ""
    return res if key.length == 0
    i = mod = (key.length % k)
    res += key[0, mod] if mod > 0
    while i < key.length
        res += (i == 0) ? "" : "-";
        res += key[i, k]
        i += k
    end
    return res.upcase
end

File: javascript/easy/licenseKeyFormatting.js
/**
 * @param {string} S
 * @param {number} K
 * @return {string}
 */
var licenseKeyFormatting = function(S, K) {
    let res = "", key = S.replace(/-/g, '');
    if (key.length == 0) return res;
    let mod = ((key.length) % K);
    if(mod > 0) {
        res += key.substring(0, mod);
    }
    while(mod < key.length) {
        res += (mod === 0) ? "" : "-";
        res += key.substring(mod, mod+K);
        mod += K;
    }
    return res.toUpperCase();
};
